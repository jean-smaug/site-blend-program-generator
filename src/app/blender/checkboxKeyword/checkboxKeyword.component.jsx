import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import _ from 'lodash';
import * as formActions from '../blender.action';

/**
 * Component for one theme's checkbox
 */

export class CheckboxKeywordComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      color: _.sample(['info', 'success', 'primary', 'warning']),
    };
  }

  toggleCheckbox = () => {
    if (!_.includes(this.props.keywords, this.props.item)) {
      this.props.addKeyword(this.props.item);
    } else this.props.removeKeyword(this.props.item);
  };

  render() {
    return (
      <span
        role="presentation"
        className={
          _.includes(this.props.keywords, this.props.item)
            ? `tag is-${this.state.color} keyword-elt`
            : 'tag is-notselected keyword-elt'
        }
        onClick={this.toggleCheckbox}
      >
        {this.props.item.charAt(0).toUpperCase() + this.props.item.substring(1).toLowerCase()}
      </span>
    );
  }
}

CheckboxKeywordComponent.propTypes = {
  addKeyword: PropTypes.func.isRequired,
  removeKeyword: PropTypes.func.isRequired,
  item: PropTypes.string.isRequired,
  keywords: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
};

const mapStateToProps = state => ({
  keywords: state.form.keywords,
});

const mapDispatchToProps = dispatch => ({
  addKeyword: (word) => {
    dispatch(formActions.addKeyword(word));
  },
  removeKeyword: (word) => {
    dispatch(formActions.removeKeyword(word));
  },
});

const CheckboxKeyword = connect(mapStateToProps, mapDispatchToProps)(CheckboxKeywordComponent);

export default CheckboxKeyword;
