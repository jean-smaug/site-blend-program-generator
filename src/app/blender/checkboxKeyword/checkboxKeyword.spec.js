import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { CheckboxKeywordComponent } from './checkboxKeyword.component';

const item = 'test';

describe('checkboxDomain.component', () => {
  let wrapper;
  let store;
  const mockStore = configureStore();
  const initialState = {};

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<CheckboxKeywordComponent item={item} store={store} />);
  });

  it('should render component', () => {
    expect(wrapper.getNodes()).toMatchSnapshot();
  });
});
