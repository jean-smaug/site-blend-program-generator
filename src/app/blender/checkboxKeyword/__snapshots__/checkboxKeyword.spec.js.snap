// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkboxDomain.component should render component 1`] = `
Array [
  <span
    className="tag is-notselected keyword-elt"
    onClick={[Function]}
    role="presentation"
>
    tech
</span>,
]
`;
